searchlib/src/vespa/searchlib/queryeval/create_blueprint_visitor_helper.cpp

```diff
@@ -66,12 +66,12 @@ template <typename WS, typename NODE>
 void
 CreateBlueprintVisitorHelper::createWeightedSet(std::unique_ptr<WS> bp, NODE &n) {
     FieldSpecList fields;
-    for (size_t i = 0; i < n.getChildren().size(); ++i) {
+    for (size_t i = 0; i < n.getNumTerms(); ++i) {
         fields.clear();
         fields.add(bp->getNextChildField(_field));
-        const query::Node &node = *n.getChildren()[i];
-        uint32_t weight = getWeightFromNode(node).percent();
-        bp->addTerm(_searchable.createBlueprint(_requestContext, fields, node), weight);
+        const auto & term = n.getAsString(i);
```

Comment:
when you have a const reference to a temporary object that contains a string reference to a mutable scratch buffer things become a bit more complicated than they seem...

Reply:
Fixed